istioctl/cmd/root.go

```diff
@@ -229,6 +225,24 @@ debug and diagnose their Istio mesh.
 	postInstallCmd.AddCommand(Webhook())
 	experimentalCmd.AddCommand(postInstallCmd)
 
+	analyzeCmd := Analyze()
+	hideInheritedFlags(analyzeCmd, "namespace", "istioNamespace")
```

Comment:
Why do we want to remove `--namespace`?  It still has a meaning for `analyze`.

For example, `istioctl analyze -n default` and `istioctl analyze -n bookinfo` are very different.

Reply:
Sorry, I misunderstood that part. Not sure if `istioNamespace` also needs to be kept? or if `hideInheritedFlags` is required for `analyze` ?

Reply:
We should keep `--istioNamespace` for analyze.